Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,251 @@
+
+One way to do this is to pull back the norm from `WithLp 1 (𝕜 × A)`, that is,
+`‖(k, a)‖ = ‖k‖ + ‖a‖` using `Unitization.addEquiv` (i.e., the identity map). However, when the norm
+on `A` is *regular* (i.e., `ContinuousLinearMap.mul`) is an isometry, there is another natural
+choice: the pullback of the norm on `𝕜 × (A →L[𝕜] A)` under the map
+`(k, a) ↦ (k, k • 1 + ContinuousLinearMap.mul 𝕜 A a)`. It turns out that among all norms on the
+unitization satisfying the properties specified above, the norm inherited from
+`WithLp 1 (𝕜 × A)` is maximal, and the norm inherited from this pullback is minimal.
+
+For possibly non-unital `RegularNormedAlgebra`s  `A` (over `𝕜`), we construct a `NormedAlgebra`
+structure on `Unitization 𝕜 A` using the pullback described above. The reason for choosing this norm
+is that for a C⋆-algebra `A` its norm is always regular, and the pullback norm on `Unitization 𝕜 A`
+is then also a C⋆-norm.
+
+## Main definitions
+
+- `Unitization.splitMul : Unitization 𝕜 A →ₐ[𝕜] (𝕜 × (A →L[𝕜] A))`: The first coordinate of this
+  map is just `Unitization.fst` and the second is the `Unitization.lift` of the left regular
+  representation of `A` (i.e., `NonUnitalAlgHom.Lmul`). We use this map to pull back the
+  `NormedRing` and `NormedAlgebra` structures.
+
+## Main statements
+
+- `Unitization.instNormedRing`, `Unitization.instNormedAlgebra`, `Unitization.instNormOneClass`,
+  `Unitization.instCompleteSpace`: when `A` is a non-unital Banach `𝕜`-algebra with a regular norm,
+  then `Unitization 𝕜 A` is a unital Banach `𝕜`-algebra with `‖1‖ = 1`.
+- `Unitization.norm_inr`, `Unitization.isometry_inr`: the natural inclusion `A → Unitization 𝕜 A`
+  is an isometry, or in mathematical parlance, the norm on `A` extends to a norm on
+  `Unitization 𝕜 A`.
+
+## Implementation details
+
+We ensure that the uniform structure, and hence also the topological structure, is definitionally
+equal to the pullback of `instUniformSpaceProd` along `Unitization.addEquiv` (this is essentially
+viewing `Unitization 𝕜 A` as `𝕜 × A`) by means of forgetful inheritance. The same is true of the
+bornology.
+
+-/
+
+variable (𝕜 A : Type*) [NontriviallyNormedField 𝕜] [NonUnitalNormedRing A]
+variable [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A]
+
+open ContinuousLinearMap
+
+namespace Unitization
+
+/-- Given `(k, a) : Unitization 𝕜 A`, the second coordinate of `Unitization.splitMul (k, a)` is
+the natural representation of `Unitization 𝕜 A` on `A` given by multiplication on the left in
+`A →L[𝕜] A`; note that this is not just `NonUnitalAlgHom.Lmul` for a few reasons: (a) that would
+either be `A` acting on `A`, or (b) `Unitization 𝕜 A` acting on `Unitization 𝕜 A`, and (c) that's a
+`NonUnitalAlgHom` but here we need an `AlgHom`. In addition, the first coordinate of
+`Unitization.splitMul (k, a)` should just be `k`. See `Unitization.splitMul_apply` also. -/
+noncomputable def splitMul : Unitization 𝕜 A →ₐ[𝕜] 𝕜 × (A →L[𝕜] A) :=
```

Comment:
I'm surprised this is non-computable!

Reply:
Hmmm... it's not. But I think I added it there originally because some IR check failed. It no longer seems to be complaining though, so I've removed `noncomputable`.